src/drivers/imu/analog_devices/adis16362/ADIS16362.cpp

```diff
@@ -0,0 +1,623 @@
+	{
+		// DIO1 output set low
+		RegisterWrite(Register::GPIO_CTRL, GPIO_CTRL_BIT::GPIO1_DIRECTION);
+		bool write0_valid = (px4_arch_gpioread(_drdy_gpio) == 1);
+
+		// DIO1 output set high
+		RegisterWrite(Register::GPIO_CTRL, GPIO_CTRL_BIT::GPIO1_DATA_LEVEL | GPIO_CTRL_BIT::GPIO1_DIRECTION);
+		bool write1_valid = (px4_arch_gpioread(_drdy_gpio) == 0);
+
+		// DIO1 output set low again
+		RegisterWrite(Register::GPIO_CTRL, GPIO_CTRL_BIT::GPIO1_DIRECTION);
+		bool write2_valid = (px4_arch_gpioread(_drdy_gpio) == 1);
+
+		if (write0_valid && write1_valid && write2_valid) {
+			PX4_INFO("DIO1 DRDY valid");
+			// Setup data ready on falling edge
+			return px4_arch_gpiosetevent(_drdy_gpio, false, true, true, &DataReadyInterruptCallback, this) == 0;
+
+		} else {
+			PX4_DEBUG("DIO1 DRDY invalid");
+		}
+	}
+
+	// check if DIO2 is connected to data ready
+	{
+		// DIO2 output set low
+		RegisterWrite(Register::GPIO_CTRL, GPIO_CTRL_BIT::GPIO2_DIRECTION);
+		bool write0_valid = (px4_arch_gpioread(_drdy_gpio) == 1);
+
+		// DIO2 output set high
+		RegisterWrite(Register::GPIO_CTRL, GPIO_CTRL_BIT::GPIO2_DATA_LEVEL | GPIO_CTRL_BIT::GPIO2_DIRECTION);
+		bool write1_valid = (px4_arch_gpioread(_drdy_gpio) == 0);
+
+		// DIO2 output set low again
+		RegisterWrite(Register::GPIO_CTRL, GPIO_CTRL_BIT::GPIO2_DIRECTION);
+		bool write2_valid = (px4_arch_gpioread(_drdy_gpio) == 1);
+
+		if (write0_valid && write1_valid && write2_valid) {
+			PX4_INFO("DIO2 DRDY valid");
+
+		} else {
+			PX4_DEBUG("DIO2 DRDY invalid");
+		}
+	}
+
+	return false;
+}
+
+bool ADIS16362::DataReadyInterruptDisable()
+{
+	if (_drdy_gpio == 0) {
+		return false;
+	}
+
+	return px4_arch_gpiosetevent(_drdy_gpio, false, false, false, nullptr, nullptr) == 0;
+}
+
+bool ADIS16362::RegisterCheck(const register_config_t &reg_cfg)
+{
+	bool success = true;
+
+	const uint16_t reg_value = RegisterRead(reg_cfg.reg);
+
+	if (reg_cfg.set_bits && ((reg_value & reg_cfg.set_bits) != reg_cfg.set_bits)) {
+		PX4_DEBUG("0x%02hhX: 0x%02hhX (0x%02hhX not set)", (uint8_t)reg_cfg.reg, reg_value, reg_cfg.set_bits);
+		success = false;
+	}
+
+	if (reg_cfg.clear_bits && ((reg_value & reg_cfg.clear_bits) != 0)) {
+		PX4_DEBUG("0x%02hhX: 0x%02hhX (0x%02hhX not cleared)", (uint8_t)reg_cfg.reg, reg_value, reg_cfg.clear_bits);
+		success = false;
+	}
+
+	return success;
+}
+
+uint16_t ADIS16362::RegisterRead(Register reg)
+{
+	set_frequency(SPI_SPEED);
+
+	uint16_t cmd[1];
+	cmd[0] = (static_cast<uint16_t>(reg) << 8);
+
+	transferhword(cmd, nullptr, 1);
+	usleep(SPI_STALL_PERIOD);
+	transferhword(nullptr, cmd, 1);
+
+	return cmd[0];
+}
+
+void ADIS16362::RegisterWrite(Register reg, uint16_t value)
+{
+	set_frequency(SPI_SPEED);
+
+	uint16_t cmd[2];
+	cmd[0] = (((static_cast<uint16_t>(reg))     | DIR_WRITE) << 8) | ((0x00FF & value));
+	cmd[1] = (((static_cast<uint16_t>(reg) + 1) | DIR_WRITE) << 8) | ((0xFF00 & value) >> 8);
+
+	transferhword(cmd, nullptr, 1);
+	usleep(SPI_STALL_PERIOD);

```

Comment:
```suggestion
	px4_udelay(SPI_STALL_PERIOD);

```

It's minor, but these should be using the new px4_udelay rather than sleep. https://github.com/PX4/PX4-Autopilot/pull/16963
In NuttX the usleep is a minimum that's realistically going to be at least a millisecond, which is substantially longer than the 9 microsecond stall time this device needs. That can make a big difference if this is sharing the SPI bus with another sensor.